### PR TITLE
feat: register denom metadata for new rollapp received ibc

### PR DESCRIPTION
**Pull request by mtsitrin**
This PR consists of the following changes:
* proto
* * added token metadata to the rollapp object
* * deprecated unused params
* cli
* * refactored `dymd tx create-rollapp` command, to load tokenmetada from json file
* IBC middleware
* * when a **new** IBC token arrives, if it belongs to a rollapp, and if a metadata is registered 
the middleware sets token metada for this IBC token in the `bank` module

# PR Standards

## Opening a pull request should be able to meet the following requirements:

---

For Author:

- [ ]  Targeted PR against correct branch
- [ ]  included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ]  Linked to Github issue with discussion and accepted design
- [ ]  Targets only one github issue
- [ ]  Wrote unit and integration tests
- [ ]  All CI checks have passed
- [ ]  Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code)

---

For Reviewer:

- [ ]  confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ]  Reviewers assigned
- [ ]  confirmed all author checklist items have been addressed

--- 

After reviewer approval:

- [ ]  In case targets main branch, PR should be squashed and merged.
- [ ]  In case PR targets a release branch, PR should be rebased.

### CODE REVIEW

**Review by omritoptix — COMMENTED** [review]
What is the rollapp denom is an ibc denom of another chain? Is it supported? e.g evmos rollapp

File: x/irc/ibc_middleware.go [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+	voucherDenom := denomTrace.IBCDenom()
+
+	// no-op if token already exist
+	if im.transferkeeper.HasDenomTrace(ctx, traceHash) {
+		return im.app.OnRecvPacket(ctx, packet, relayer)
+	}
+
+	rollapp, found := im.rollappkeeper.GetRollapp(ctx, rollappID)
+	if !found {
+		panic("handling dymint IBC transfer packet for non-existing rollapp")
```
[the remark this answers is not in the document]
Comment (mtsitrin) [reply x/irc/ibc_middleware.go:153]: `rollappkeeper.GetRollapp` called on the UpdateClient validation as well.
logic wise, we can't reach this point, that way IMO panicking is fine.
Better to have DoS than exploitation
WDYT

**Comment by mtsitrin** [comment]
> What is the rollapp denom is an ibc denom of another chain? Is it supported? e.g evmos rollapp
The metadata won't be created in this case, as it's a token that originated from the hub, so it's not creating a new IBC token.

AFAIK, cosmos chains doesn't register it's IBC connected partners in their own bank metadata
u can see osmosis and juno for example. no IBC tokens, only custom stuff
https://lcd.osmosis.zone/cosmos/bank/v1beta1/denoms_metadata
https://rest.cosmos.directory/juno/cosmos/bank/v1beta1/denoms_metadata

File: x/irc/ibc_middleware.go [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+	denomTrace := transfertypes.ParseDenomTrace(prefixedDenom)
+	traceHash := denomTrace.Hash()
+	voucherDenom := denomTrace.IBCDenom()
+
+	// no-op if token already exist
+	if im.transferkeeper.HasDenomTrace(ctx, traceHash) {
+		return im.app.OnRecvPacket(ctx, packet, relayer)
+	}
+
+	rollapp, found := im.rollappkeeper.GetRollapp(ctx, rollappID)
```
[the remark this answers is not in the document]
Comment (mtsitrin) [reply x/irc/ibc_middleware.go:151]: wdym 'initial checks'?

**Review by omritoptix — APPROVED** [review]

**Review by omritoptix — APPROVED** [review]

**Pull request merged**